src/clientdlg.cpp

```diff
@@ -608,11 +603,8 @@ void CClientDlg::closeEvent ( QCloseEvent* Event )
     ConnectDlg.close();
     AnalyzerConsole.close();
 
-    // if connected, terminate connection
-    if ( pClient->IsRunning() )
-    {
-        pClient->Stop();
-    }
+    // Disconnect if needed
+    pClient->Stop();
 
     // make sure all current fader settings are applied to the settings struct
     MainMixerBoard->StoreAllFaderSettings();
```

Comment:
Maybe `closeEvent` should just call `QCoreApplication::instance()->exit();` and then do the work in `OnAboutToQuit` handlers?  That should stop needing to call `XYZ.Stop()`.

Reply:
https://github.com/jamulussoftware/jamulus/issues/3402